feat: Add request items feature with UI and repository

BarreiroReSird opened the pull request:
Introduces the request items flow, including domain model, repository, UI state, ViewModel, and Compose UI components for listing, filtering, and selecting products. Updates theme colors to support new UI elements. Also adds a TODO in PendingRequestsRepository for future database integration.

BarreiroReSird committed:
feat: Add request items feature with UI and repository

Introduces the request items flow, including domain model, repository, UI state, ViewModel, and Compose UI components for listing, filtering, and selecting products. Updates theme colors to support new UI elements. Also adds a TODO in PendingRequestsRepository for future database integration.

BarreiroReSird commented:
@Basiiii

Basiiii commented on app/app/src/main/java/com/lojasocial/app/repository/PendingRequestsRepository.kt:
```diff
@@ -9,6 +9,7 @@ import javax.inject.Inject
 
 class PendingRequestsRepository @Inject constructor() {
     fun getPendingRequests(): Flow<List<PendingRequest>> = flow {
+        // TODO: Conectar na base de dados
```
O PR tem que ter a funcionalidade completamente implementada, não podes só meter TODO

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/SearchBar.kt:
```diff
@@ -0,0 +1,43 @@
+        ),
+        singleLine = true
+    )
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/RequestStatusHeader.kt:
```diff
@@ -0,0 +1,70 @@
+            trackColor = ProgressFull,
+        )
+    }
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/RequestItemsTopBar.kt:
```diff
@@ -0,0 +1,41 @@
+            containerColor = MaterialTheme.colorScheme.background
+        )
+    )
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/RequestItemsTopAppBar.kt:
```diff
@@ -0,0 +1,41 @@
+            containerColor = MaterialTheme.colorScheme.background
+        )
+    )
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/RequestItemsBottomBar.kt:
```diff
@@ -0,0 +1,49 @@
+            Text("Submeter Pedido", fontSize = 18.sp, fontWeight = FontWeight.SemiBold)
+        }
+    }
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/ProductItemRow.kt:
```diff
@@ -0,0 +1,131 @@
+            )
+        }
+    }
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/requestitems/components/CategoryFilters.kt:
```diff
@@ -0,0 +1,53 @@
+            }
+        }
+    }
+}
```
Falta preview

Basiiii commented on app/app/src/main/java/com/lojasocial/app/repository/RequestItemsRepository.kt:
```diff
@@ -0,0 +1,18 @@
+
+import com.lojasocial.app.domain.RequestItem
+
+open class ProductsRepository {
```
Implementa ligação à base dados

BarreiroReSird committed:
refactor: Refactor request items UI and repository integration

Replaces ProductsRepository with RequestItemsRepository using Firestore for data retrieval. Adds @Preview composables for UI components to improve testing and visualization. Removes the redundant RequestItemsTopAppBar file and corrects the package name in RequestItemsTopBar.

BarreiroReSird committed:
refactor: Refactor RequestItems UI and ViewModel structure

Refactored RequestItemsView to extract RequestItemsContent for improved composability and preview support. Updated ViewModel to use RequestItemsRepository instead of ProductsRepository. Enhanced RequestStatusHeader with animated progress and improved styling. Added multiple UI previews for different states and cleaned up unused imports.

BarreiroReSird committed:
refactor: Integrate Hilt DI into RequestItemsViewModel and navigation

Refactored RequestItemsViewModel to use @HiltViewModel and constructor injection for RequestItemsRepository. Updated RequestItemsView to use hiltViewModel(). Added navigation from BeneficiaryPortalView to RequestItemsView via QuickActionsSection, and passed navigation callbacks through the relevant composables.

BarreiroReSird commented:
@Basiiii

BarreiroReSird committed:
refactor: implement repository pattern and DI for request items flow

Introduce a DI module for repository binding and refactor the RequestItem model to use Firestore document IDs with updated fields. Implement the repository interface pattern with a Firestore-backed implementation and update the ViewModel and UI accordingly for improved product ID handling, submission state management, and category constants. Enhance the flow with submission logic, UI disablement during submission, and better preview and error handling support.

BarreiroReSird committed:
fix: ensure navigation triggers on submission error in RequestItemsView

Update RequestItemsView to call onSubmitClick, guaranteeing consistent navigation behavior after any submission attempt. Includes minor code cleanups and formatting adjustments.

BarreiroReSird commented:
@Basiiii

BarreiroReSird committed:
refactor: split repository responsibilities and update dependencies

Split the monolithic RequestItemsRepository into separate ItemsRepository (for product retrieval) and OrdersRepository (for order submission) to improve separation of concerns. Update the dependency injection bindings and refactor the RequestItemsViewModel to depend on the new repository structure, enhancing code maintainability and architectural clarity.

BarreiroReSird committed:
feat: implement pagination with lazy loading for product requests

Introduce pagination support in ItemsRepository and RequestItemsViewModel to fetch products in pages. Enable lazy loading in RequestItemsView, loading more items automatically as the user scrolls. Includes refactoring of DI module naming, updates to RequestItem model to support mutable document IDs, and adjustments to UI previews and logic to accommodate the new paginated data flow.

BarreiroReSird committed:
refactor: align Firestore collection naming with data model

Update OrdersRepositoryImpl to use the 'requests' collection instead of 'applications', and rename the corresponding data map keys. This change ensures the codebase is consistent with the updated data model terminology.

BarreiroReSird committed:
refactor: separate repository interfaces from their implementations

Move the concrete implementation classes ItemsRepositoryImpl and OrdersRepositoryImpl into their own dedicated files, while keeping only the interface definitions in ItemsRepository.kt and OrdersRepository.kt. This improves code organization, enhances separation of concerns, and increases modularity.

BarreiroReSird commented:
@Basiiii

BarreiroReSird committed:
Merge branch 'main' into feature/implement-request-items

BarreiroReSird committed:
refactor: improve navigation structure for beneficiary quick actions

Update QuickActionsSection and BeneficiaryPortalView to use explicit navigation callbacks for orders and pickups. Adjust the navigation routing in MainActivity to correctly direct order actions to RequestItemsView. Includes cleanup of unused imports and removal of obsolete preview configurations.
